sdk/mx.sdk/mx_sdk_vm.py

```diff
@@ -655,13 +667,16 @@ def jlink_new_jdk(jdk, dst_jdk_dir, module_dists,
                 mx.logv('[Creating JDK without -XX:ThreadPriorityPolicy=1]')
                 thread_priority_policy_option = ''
 
-            if any((m.name == 'jdk.internal.vm.compiler' for m in modules)):
-                jlink.append('--add-options=-XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCIProduct -XX:-UnlockExperimentalVMOptions' + thread_priority_policy_option)
+            if jdk_supports_enablejvmciproduct(jdk):
+                if any((m.name == 'jdk.internal.vm.compiler' for m in modules)):
+                    jlink.append('--add-options=-XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCIProduct -XX:-UnlockExperimentalVMOptions' + thread_priority_policy_option)
+                else:
+                    # Don't default to using JVMCI as JIT unless Graal is being updated in the image.
+                    # This avoids unexpected issues with using the out-of-date Graal compiler in
+                    # the JDK itself.
+                    jlink.append('--add-options=-XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCIProduct -XX:-UseJVMCICompiler -XX:-UnlockExperimentalVMOptions' + thread_priority_policy_option)
             else:
-                # Don't default to using JVMCI as JIT unless Graal is being updated in the image.
-                # This avoids unexpected issues with using the out-of-date Graal compiler in
-                # the JDK itself.
-                jlink.append('--add-options=-XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCIProduct -XX:-UseJVMCICompiler -XX:-UnlockExperimentalVMOptions' + thread_priority_policy_option)
+                mx.logv('[Creating JDK without -XX:+EnableJVMCIProduct]')
```

Comment:
You need the following to make ThreadPriorityPolicy still be applied when EnableJVMCIProduct is not supported :
```
if thread_priority_policy_option:
    jlink.append('--add-options=' + thread_priority_policy_option.strip())
```

Reply:
Right, sorry about that. Should be fixed now. More thoughts?

Reply:
looks good

Reply:
Thanks very much for the review, Doug!